### ISSUE TITLE
🥹 Base Mainnet Deployment

### ISSUE THREAD

**Issue by pcaversaccio** [pcaversaccio/xdeployer#164]
**TL;DR:** Unfortunately, it is not possible to deploy the [`Create2Deployer`](https://github.com/pcaversaccio/create2deployer/blob/main/contracts/Create2Deployer.sol) contract on the Base production network.

### Background

When [bridging](https://etherscan.io/tx/0x1aca963c67fd43800a0e298f8c93c0a2e373d0631339cfdebc5937d568a01b62) the funds to the Base mainnet, Base applied a recently introduced (but unfortunately unexpected) behaviour in the Bedrock upgrade that essentially sends a [`self`](https://basescan.org/tx/0xece86dacdfdd700e7bd5551630b00432280afe09325417f435ca46e50f710daf) transfer on L2. I.e. the nonce of the deployer account [`0x554282cf65b42fc8fddc6041eb24ce5e8a0632ad`](https://basescan.org/address/0x554282Cf65b42fc8fddc6041eb24ce5E8A0632Ad) was increased by 1 before deploying the [`Create2Deployer`](https://github.com/pcaversaccio/create2deployer/blob/main/contracts/Create2Deployer.sol) contract. Unfortunately, this means that I can no longer deploy the contract to `0x13b0D85CcB8bf860b6b79AF3029fCA081AE9beF2`. If the demand for this chain is high enough, I might consider redeploying the [`Create2Deployer`](https://github.com/pcaversaccio/create2deployer/blob/main/contracts/Create2Deployer.sol) contract to a new address including the Base mainnet.

![image](https://github.com/pcaversaccio/create2deployer/assets/25297591/cbd73879-d2a5-45f0-a5fb-80e80e7dad99)

**Comment by nebolax** [comment]
Ouffffff this is so sad :((( What would be the alternatives?

**Comment by pcaversaccio** [comment]
> Ouffffff this is so sad :((( What would be the alternatives to how to deploy on Base?

Well I got good news for you: https://github.com/ethereum-optimism/op-geth/pull/126. It will be added as predeploy to Base. This might take however until January most probably.

**Comment by nebolax** [comment]
By the way, question. If you will include an option to `selfdestruct` in the new version of `Create2Deployer`, this would allow in case of such problems to destroy the contract and re-deploy it, right?

Would it pose any security risks? I guess it would not, right?

Then maybe you could add it in the new version so that if similar problems happen in the future, they are not so huge.

**Comment by pcaversaccio** [comment]
> By the way, question. If you will include an option to `selfdestruct` in the new version of `Create2Deployer`, this would allow in case of such problems to destroy the contract and re-deploy it, right?
> 
> Would it pose any security risks? I guess it would not, right?
> 
> Then maybe you could add it in the new version so that if similar problems happen in the future, they are not so huge.

There will be no `selfdestruct` nor any other governance function in the new `Create2Deployer`! If you wanna have a look at the new version, here you go: https://github.com/pcaversaccio/createx. It's muuuuuch more powerful now and will also be available on Base Mainnet.

**Comment by pcaversaccio** [comment]
As of `xdeployer` [`v3.0.0`](https://github.com/pcaversaccio/xdeployer/releases/tag/v3.0.0), `baseMain` deployments are possible 🍻!

**Comment by nebolax** [comment]
Awesome!!!!!!!!!! ❤️
